### PR TITLE
Jms connector SE api 

### PR DESCRIPTION
**Pull request by danielkec**
Add basic support for JMS connector usage in SE flavor

### CODE REVIEW

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java [hunk 1]
```diff
@@ -122,6 +122,34 @@
     private final ExecutorService executor;
     private final Map<String, SessionMetadata> sessionRegister = new HashMap<>();
 
+    /**
+     * Creates a new instance of JmsConnector with the required configuration.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     * @return the new instance
+     */
+    public static JmsConnector create(io.helidon.config.Config config) {
```
Comment (tomas-langer) [inline messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java:131]: This class should use a builder. If we need a separate constructor for CDI, so be it, but there should be a constructor with a builder, that allows explicit configuration of everything that can be configured through config.
In SE, there is always a choice for developer - either build everything by hand, or use a config, or a combination

For example the executor services must be configurable, as user may want to reuse an executor service for multiple components

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java (hunk 1 above)
Comment (danielkec) [reply messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java]: Great idea, fixed

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java [hunk 2]
```diff
@@ -116,20 +116,24 @@
...
+     * Provides a {@link io.helidon.messaging.connectors.jms.JmsConnectorBuilder} for creating
+     * a {@link io.helidon.messaging.connectors.jms.JmsConnector} instance.
      *
-     * @param config Helidon {@link io.helidon.config.Config config}
-     * @return the new instance
+     * @return new Builder instance
      */
-    public static JmsConnector create(io.helidon.config.Config config) {
-        return new JmsConnector(config, null);
+    public static JmsConnectorBuilder builder() {
```
Comment (tomas-langer) [inline messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java:135]: The `Builder` should be a static inner class of `JmsConnector`, not a standalone class - is there a reason for breaking this rule?

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnectorBuilder.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        return this;
+    }
+
+    /**
+     * Custom executor for asynchronous operations like acknowledgement.
+     *
+     * @param executor custom executor service
+     * @return this builder
+     */
+    public JmsConnectorBuilder executor(ExecutorService executor) {
```
Comment (tomas-langer) [inline messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnectorBuilder.java:71]: Please also add methods for `Supplier<ExecutorService>` (and `Supplier<ScheduledExecutorServer>` below) so we can pass our `ThreadPoolSupplier` directly.

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java (hunk 2 above)
Comment (danielkec) [reply messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java:136]: no reason, fixed

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnectorBuilder.java (hunk 3 above)
Comment (danielkec) [reply messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnectorBuilder.java]: added

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: messaging/connectors/aq/src/main/java/io/helidon/messaging/connectors/aq/AqConnector.java [hunk 4]
```diff
@@ -12,152 +12,172 @@
...
-    @Override
-    protected JmsMessage<?> createMessage(javax.jms.Message message,
-                                          Executor executor,
-                                          SessionMetadata sessionMetadata) {
-        return new AqMessageImpl<>(super.createMessage(message, executor, sessionMetadata), sessionMetadata);
-    }
+        @Override
+        public AqConnectorImpl build() {
+            if (config == null) {
+                config = io.helidon.config.Config.create();
```
[the remark this answers is not in the document]
Comment (danielkec) [reply messaging/connectors/aq/src/main/java/io/helidon/messaging/connectors/aq/AqConnector.java:162]: That was the idea, connector without any configuration is useless

File: dependencies/pom.xml [hunk 5]
```diff
@@ -79,6 +79,7 @@
         <version.lib.jsonb-api>1.0.2</version.lib.jsonb-api>
         <version.lib.jsonp-api>1.1.6</version.lib.jsonp-api>
         <version.lib.jsonp-impl>1.1.6</version.lib.jsonp-impl>
+        <version.lib.jta>1.1</version.lib.jta>
```
[the remark this answers is not in the document]
Comment (danielkec) [reply dependencies/pom.xml:82]: yay thx thats leftover

File: messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java [hunk 6]
```diff
@@ -116,11 +117,43 @@
...
 
+    static final String SCHEDULER_THREAD_NAME_PREFIX = "jms-poll-";
+    static final String EXECUTOR_THREAD_NAME_PREFIX = "jms-";
+
     private final Instance<ConnectionFactory> connectionFactories;
 
     private final ScheduledExecutorService scheduler;
     private final ExecutorService executor;
     private final Map<String, SessionMetadata> sessionRegister = new HashMap<>();
+    private Map<String, ConnectionFactory> connectionFactoryMap;
```
[the remark this answers is not in the document]
Comment (danielkec) [reply messaging/connectors/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsConnector.java]: fixed

**Review by tomas-langer — APPROVED** [review]
LGTM

File: messaging/connectors/aq/src/main/java/io/helidon/messaging/connectors/aq/AqConnector.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomas-langer) [reply messaging/connectors/aq/src/main/java/io/helidon/messaging/connectors/aq/AqConnector.java]: We have agreed to use empty config, as that is the expectation by user. `Config.create()` may pick up unexpected configuration files.

**Pull request merged**